lib/index.js

```diff
@@ -28,26 +43,53 @@ NamedURLResolverClass.prototype.escape = function(string) {
 
 NamedURLResolverClass.prototype.resolve = function(name, params) {
     if(name && (name in this.routesMap)) {
-        if(!params) params = {};
-
         var routePath = this.routesMap[name];
-        for(var paramName in params) {
-            if(params.hasOwnProperty(paramName)) {
-                let paramRegex = new RegExp('(/|^):' + paramName + '(/|$)');
-                let paramValue = this.escape('' + params[paramName]);
-                routePath = routePath.replace(paramRegex, '$1' + paramValue + '$2');
+
+        if(!params) {
+            return routePath;
+        }
+
+        for (var paramName in params) {
+            if (params.hasOwnProperty(paramName)) {
+                var paramValue = params[paramName];
+        
+                if (paramName === "splat") { // special param name in RR, used for "*" and "**" placeholders
+                    paramValue = toArray(paramValue); // when there are multiple globs, RR defines "splat" param as array.
+                    var i = 0;
+                    routePath = routePath.replace(reSplatParams, (match) => {
+                        var val = paramValue[i++];
+                        // don't escape slashes for double star, as "**" considered greedy by RR spec
+                        return val == null ? "" : this.escape('' + val, match === "*");
+                    });
+                } else {
+                    // Rougly resolve all named placeholders.
+                    // Cases: 
+                    // - "/path/:param"
+                    // - "/path/(:param)"
+                    // - "/path(/:param)"
+                    // - "/path(/:param/):another_param"
+                    // - "/path/:param(/:another_param)"
+                    // - "/path(/:param/:another_param)"
+                    var paramRegex = new RegExp('(\/|\\(|\\)|^):' + paramName + '(\/|\\)|\\(|$)');
+                    var replacement = '$1' + this.escape('' + paramValue) + '$2';
+                    routePath = routePath.replace(paramRegex, replacement);
+                }
             }
         }
-        return routePath;
+        return routePath
+            // Remove braces around resolved optional params (i.e. "/path/(value)")
```

Comment:
Actually if those parens can be nested, things are getting ugly in the following case:

```js
<Route name="user_edit" path="/user(/:teamId(/:projectId(/:userId)))" />
resolver.resolve("url", {teamId: 1, userId: 2});
```

This is a hard problem to solve with regexps since JS lacks recursive Regexps and there is a lot of different ways to structure those optional parameters, e.g. this one `/user(/:p1(/:p2)(/:p3(/:p4)(/:p5)))`. Maybe hours of work with some sophisticated back-references would help but anyway since this is a tree, a proper parser would have to be involved. So a proper solution would be an overkill.

Let's just go with the `@` way you proposed and we'll have to live with the fact that there is an unhandled corner-case, so what if it works in most of the sane use-cases. We'll think of a way to properly solve this if this package is more widely adopted some day.